### ISSUE TITLE
Unable to start Jenkins because of base plugins dependency

### ISSUE THREAD

**Issue by eirriel** [jenkinsci/kubernetes-operator#704]
**Describe the bug**
Jenkins not starting because of update of base plugin is required but operator doesn't accept the new format on version number

Operator:
```
controller-jenkins invalid plugin version 'workflow-job:1167.v8fe861b_09ef9', must follow pattern '^[0-9a-zA-Z+\\.-]+$' {"cr": "jenkins"}
github.com/go-logr/zapr.(*zapLogger).Info
```

Jenkins:
```
2022-01-28 00:43:30.300+0000 [id=29] SEVERE jenkins.InitReactorRunner$1#onTaskFailed: Failed Loading plugin Pipeline Graph Analysis Plugin v188.v3a01e7973f2c (pipeline-graph-analysis)
java.io.IOException: Failed to load: Pipeline Graph Analysis Plugin (188.v3a01e7973f2c)
- Update required: Pipeline: Job (2.42) to be updated to 1145.v7f2433caa07f or higher

```
Kubernetes version: AWS EKS v1.19.6-eks-49a6c0
Jenkins Operator version: 0.7.0

Thank you in advance!!

**Comment by lemeurherve** [comment]
Should be resolved when https://github.com/jenkinsci/kubernetes-operator/pull/699 is released.

**Comment by manut** [comment]
Would be great to have a container image with the fix included available at:
https://hub.docker.com/r/virtuslab/jenkins-operator/tags

Or is there a new location to look for current prebuilt images?

**Comment by DionJones615** [comment]
@lemeurherve Can we get a release or a work-around?  I don't see any way to work around this, so I cannot deploy a new jenkins instance until I have the new release or build the container image myself.

**Comment by Daklon** [comment]
I'm facing this same issue, as a workaround you can try to use an older jenkins docker image version, I'm using this one and is working for me
`image: jenkins/jenkins:2.303.3-lts-alpine`

**Comment by Daklon** [comment]
Found another workaround that will also allow us to use the latest supported lts jenkins, just install an older version of the problematic plugin:
`apiVersion: jenkins.io/v1alpha2
kind: Jenkins
metadata:
  name: jenkins
  namespace: jenkins
spec:
   master:
        plugins:
        - name: pipeline-graph-analysis
          version: "1.12"
        containers:
            - name: jenkins-master
              image: jenkins/jenkins:lts-alpine`

**Comment by lemeurherve** [comment]
@DionJones615 I can't do anything as this plugin is release via a Github Action, not via Jenkins Infrastructure.

**Comment by Sig00rd** [comment]
Hello everyone, thanks for reporting this. The snapshot release that is available now (`virtuslab/jenkins-operator:29c5b1e`) should have this fixed.

**Comment by cbernard-psee** [comment]
@Sig00rd  Is there a plan to release a v0.8 shortly to include this fix? Many jenkins plugins start to follow this new pattern

**Comment by brokenpip3** [comment]
Yeah second that ^, would be great to have a new release :) 
thanks!

**Comment by thecooldrop** [comment]
Why does the operator even validate the versions? It would be much simpler for both users and the operator to fail with an "Plugin not found" type of error in case of invalid versions. This would make it very obvious what went wrong.

**Comment by snwbr** [comment]
Is it only me, or is anyone else getting reverted back to `jenkins-operator:v0.7.0` when using `jenkins-operator:29c5b1e`? This for some reason also reverts back Jenkins from `jenkins:2.332.1-lts-alpine` to `jenkins:2.319.1-lts-alpine`

**Comment by thecooldrop** [comment]
What do you mean by "reverted back" ?

**Comment by snwbr** [comment]
> What do you mean by "reverted back" ?

Well, that :) when doing `kubectl describe pod jenkins-operator-xxx` it showed `jenkins-operator:29c5b1e` as the image. Suddenly out of nothing, a new operator starts with the old image `v0.7.0` and so does `jenkins` controller. I've checked and no old replicasets are there.

**Comment by thecooldrop** [comment]
Hi  @snwbr, thanks for quick response. That is really weird behavior but seems to be a Kubernetes issue, rather than operator issue, since the operator does not manage itself. Can you reproduce such behavior in local cluster ( in kind or minikube ) ? If yes could you please provide the steps necessary to reproduce the issue ?

**Comment by snwbr** [comment]
> Hi @snwbr, thanks for quick response. That is really weird behavior but seems to be a Kubernetes issue, rather than operator issue, since the operator does not manage itself. Can you reproduce such behavior in local cluster ( in kind or minikube ) ? If yes could you please provide the steps necessary to reproduce the issue ?

I'm quite sure it's K8s yeah... just wanted to know if it's a default behavior. Since I'm working on an environment that can be destroyed and recreated, I did so and it seems this time is OK. So probably there was some old replica set or something... Thanks @thecooldrop!

**Comment by brokenpip3** [comment]
closing due to everything was fixed in master and the new `0.8.0-beta`, let me know if it's not like that and you will still facing this issue with that version. Thanks
